client/src/components/DisplayTable/DisplayTable.js

```diff
@@ -174,6 +193,38 @@ export class DisplayTable extends React.Component {
       >
         {!disable_header_utils && (
           <div className={"display-table-container__utils"}>
+            <DisplayTableDropdownFilter
+              columns={
+                <LegendList
+                  items={_.map(all_ordered_col_keys, (key) => ({
+                    id: key,
+                    label: col_configs_with_defaults[key].header,
+                    active: _.includes(visible_ordered_col_keys, key),
+                  }))}
+                  onClick={(clicked_key) => {
+                    !(
+                      visible_ordered_col_keys.length === 1 &&
+                      visible_ordered_col_keys.includes(clicked_key)
+                    ) &&
+                      this.setState({
+                        visible_ordered_col_keys: _.chain(
+                          col_configs_with_defaults
+                        )
+                          .pickBy((col, key) =>
+                            _.chain(visible_ordered_col_keys)
+                              .toggle_list(clicked_key)
+                              .includes(key)
+                              .value()
+                          )
+                          .map(({ index }, key) => [index, key])
+                          .sortBy(_.first)
+                          .map(_.last)
+                          .value(),
```

Comment:
This would be made a lot simpler if the array of visible keys was stored in state un-ordered, e.g. as just `visible_col_keys`. `all_ordered_col_keys` is declared during render, so the ordered version of visible col keys may as well also be calculated (e.g. `_.without(all_ordered_col_keys, visible_col_keys)`) rather than stored in state. Then updating the state would be as easy as `_.toggle_list(visible_col_keys, clicked_key)`.

Not the most common/direct use of it, but this is sort of a single-source-of-truth thing. Column order is derived from `col_configs_with_defaults` within the render function, so encoding it back in to state with `visible_ordered_col_keys` is just an inefficient duplication.